instrumentation/ktor/ktor-3.0/testing/src/main/kotlin/io/opentelemetry/instrumentation/ktor/v3_0/client/KtorHttpClientSingleConnection.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.ktor.v3_0.client
+
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.request.*
+import io.opentelemetry.instrumentation.testing.junit.http.SingleConnection
+import kotlinx.coroutines.runBlocking
+
+class KtorHttpClientSingleConnection(
+  private val host: String,
+  private val port: Int,
+  private val installTracing: HttpClientConfig<*>.() -> Unit,
+) : SingleConnection {
+
+  private val client: HttpClient
+
+  init {
+    val engine = CIO.create {
```

Comment:
Please make sure we're closing engines created with the `create` function` manually.

You could also use something like:
```kotlin
HttpClient(CIO) {
    engineConfig {
        maxConnectionsCount = 1
    }
    
    installTracing()
}
```
In this case, client will manage engine for you

Reply:
couldn't make this work because `engineConfig` is internal

Reply:
Sorry, my bad. It is:
```kotlin
engine {
}
```, not `engineConfig`.

Reply:
thanks, change made